x-pack/platform/plugins/shared/ml/public/application/components/job_selector/group_or_job_selector_menu/group_selector_menu.tsx

```diff
@@ -0,0 +1,116 @@
+import { EuiContextMenu, EuiNotificationBadge, EuiPopover } from '@elastic/eui';
+
+import React, { useMemo, useState } from 'react';
+import { EuiButton } from '@elastic/eui';
+import { useUrlState } from '@kbn/ml-url-state';
+import type { MlPages } from '../../../../locator';
+import { useJobInfoFlyouts } from '../../../jobs/components/job_details_flyout/job_details_flyout_context';
+import { getOptionsForJobSelectorMenuItems } from './get_options_for_job_selector_menu';
+import { useMlKibana } from '../../../contexts/kibana';
+
+export const GroupSelectorMenu = ({
+  groupId,
+  jobIds,
+  page,
+  onRemoveJobId,
+  removeJobIdDisabled,
+}: {
+  groupId: string;
+  jobIds: string[];
+  page: MlPages;
+  onRemoveJobId: (jobId: string) => void;
+  removeJobIdDisabled: boolean;
+}) => {
+  const [isPopoverOpen, setIsPopoverOpen] = useState(false);
+  const { setActiveFlyout, setActiveJobId } = useJobInfoFlyouts();
+  const {
+    services: {
+      share,
+      application: { navigateToUrl },
+    },
+  } = useMlKibana();
+  const [globalState] = useUrlState('_g');
+
+  const closePopover = () => setIsPopoverOpen(false);
+  const onButtonClick = () => setIsPopoverOpen(true);
+
+  const popoverId = `mlAnomalyGroupPopover-${groupId}`;
+  const button = (
+    <EuiButton
+      iconType="folderClosed"
+      iconSide="left"
+      onClick={onButtonClick}
+      size="s"
+      color="text"
+    >
+      {groupId}
+      <EuiNotificationBadge size="s">{jobIds.length}</EuiNotificationBadge>
+    </EuiButton>
+  );
+  const panels: EuiContextMenuPanelDescriptor[] = useMemo(() => {
+    const items: EuiContextMenuPanelDescriptor[] = [
+      {
+        id: 0,
+        items: jobIds.map((jobId, idx) => ({
+          panel: jobId,
+          name: jobId,
+        })),
+      },
+    ];
+
+    jobIds.forEach((jobId, idx) => {
+      const options = getOptionsForJobSelectorMenuItems({
+        jobId,
+        page,
+        onRemoveJobId,
+        removeJobIdDisabled,
+        showRemoveJobId: false,
+        closePopover,
+        globalState,
+        setActiveFlyout,
+        setActiveJobId,
+        navigateToUrl,
+        share,
+      });
+      const jobPanel: EuiContextMenuPanelDescriptor = {
+        id: jobId,
+        title: jobId,
+        items: options,
+      };
+      items.push(jobPanel);
+    });
+    return items;
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [
+    jobIds,
+    setActiveFlyout,
+    setActiveJobId,
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    JSON.stringify(globalState),
+    navigateToUrl,
+    share,
+    page,
+    onRemoveJobId,
+    removeJobIdDisabled,
+  ]);
+  return (
+    <EuiPopover
+      id={popoverId}
+      button={button}
+      isOpen={isPopoverOpen}
+      closePopover={closePopover}
+      panelPaddingSize="none"
+      anchorPosition="downLeft"
+    >
+      <EuiContextMenu initialPanelId={0} panels={panels} />
```

Comment:
For the Group menu, is it possible to add a `Remove from Anomaly Explorer` item to remove the whole group from the view in a single click?

Reply:
Updated here [`466ddec` (#207141)](https://github.com/elastic/kibana/pull/207141/commits/466ddec987d94e12d4afef7df5b5f17fd4a85ea8)

Reply:
If I have 1 group and 1 or more jobs selected for display in the view, the `Remove group from explorer` is disabled. Can it be enabled as if there are other groups _or_ jobs being displayed?